File: bundle/src/main/java/com/example/aem/bundle/HelloService.java
package com.example.aem.bundle;

import com.example.aem.bundle.services.posts.Post;
import com.example.aem.bundle.services.posts.PostsService;
import org.osgi.service.component.annotations.Activate;
import org.osgi.service.component.annotations.Component;
import org.osgi.service.component.annotations.Reference;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.List;

/**
 * Note that the site configuration is injected, not retrieved.
 */
@Component(service = HelloService.class, immediate = true)
public class HelloService {

    private static final Logger LOG = LoggerFactory.getLogger(HelloService.class);

    private PostsService postsService;

    @Reference
    void setPostsService(PostsService postsService) {
        this.postsService = postsService;
    }

    @Activate
    protected void activate() {
        final List<Post> posts = postsService.getPosts();

        LOG.info("Posts service contains knowledge of {} post(s).", posts.size());
    }

}

File: bundle/src/main/kotlin/com.example.aem.bundle/HelloService.kt
package com.example.aem.bundle

import com.example.aem.bundle.services.posts.PostsService
import org.osgi.service.component.annotations.Activate
import org.osgi.service.component.annotations.Component
import org.osgi.service.component.annotations.Reference
import org.slf4j.LoggerFactory

/**
 * Note that the site configuration is injected, not retrieved.
 */
@Component(service = arrayOf(HelloService::class), immediate = true)
class HelloService {

    companion object {
        private val LOG = LoggerFactory.getLogger(HelloService::class.java)
    }

    @Reference
    private lateinit var postsService: PostsService

    @Activate
    protected fun activate() {
        val posts = postsService.posts

        LOG.info("Posts service contains knowledge of {} post(s).", posts.size)
    }

}
